### PR TITLE
Feature/ees 1219 set publication methodology

### PR DESCRIPTION
**Pull request by jamescutts**
This PR allows setting of a publications methodology to either a External Link or Internal Methodology

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs [hunk 1]
```diff
@@ -1,12 +1,14 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs]: This doesn't appear to be used

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 using System.Linq;
 using System.Threading.Tasks;
 using AutoMapper;
 using GovUk.Education.ExploreEducationStatistics.Admin.Controllers.Utils;
 using GovUk.Education.ExploreEducationStatistics.Admin.Models.Api;
 using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
 using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Admin.Validators;
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs]: This import doesn't appear to be used

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs [hunk 3]
```diff
@@ -86,7 +90,35 @@ public class PublicationService : IPublicationService
...
                 .OnSuccess(publication => _mapper.Map<PublicationViewModel>(publication));
         }
-        
+
+        public async Task<Either<ActionResult, bool>> UpdatePublicationMethodologyAsync(Guid publicationId,
+            UpdatePublicationMethodologyViewModel methodology)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Publication>(publicationId)
+                //.OnSuccess(_userService.CheckCanUpdatePublicationMethodology)
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs]: This line looks like it's redundant now

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs [hunk 4]
```diff
@@ -95,5 +127,35 @@ public static IQueryable<Publication> HydratePublicationForPublicationViewModel(
...
+            {
+                var methodology = await _context.Methodologies.FirstOrDefaultAsync(p => p.Id == model.MethodologyId);
+
+                if (methodology == null)
+                {
+                    return ValidationActionResult(MethodologyDoesNotExist);
+                }
+
+                // TODO: this will want updating when methodology status is fully implemented
+                if (methodology.Status != "Draft")
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs]: This isn't going to work assigning seeded methodologies which already have a status of Live.

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs (hunk 3 above)
Comment (jamescutts) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs]: Placeholder for methodology permission work

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs (hunk 4 above)
Comment (jamescutts) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs]: Good spot, inverted logic and updated tests accordingly

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs [hunk 5]
```diff
@@ -97,21 +97,26 @@ public class PublicationService : IPublicationService
                 .OnSuccess(_ => ValidateMethodologyCanBeAssigned(methodology))
                 .OnSuccess(async _ =>
                 {
-                    var publication = await _context.Publications.Include(p => p.Methodology)
+                    var publication = await _context.Publications.Include(p => p.ExternalMethodology).Include(p => p.Methodology)
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs]: Do you need to include Methodology?

**Pull request merged**